components/openvsx-proxy/pkg/handler.go

```diff
@@ -49,7 +52,6 @@ func (o *OpenVSXProxy) Handler(p *httputil.ReverseProxy) func(http.ResponseWrite
 		key, err := o.key(r)
 		if err != nil {
 			log.WithFields(logFields).WithError(err).Error("cannot create cache key")
-			r.Host = o.upstreamURL.Host
```

Comment:
it seems to break: https://github.com/gitpod-io/gitpod/pull/12071#issuecomment-1212756291

Do we really need to change it? OpenVSX should care about `X-Forwarded*` to create links or it also looks at Host header for it?

Reply:
It is interesting though in context of Self Hosted in air gapped scenario where maybe a user machine does not have access to open-vsx.org directly but only via our proxy. In this case it would be even desirable. Not sure. cc @corneliusludmann